src/chat/dtos/chat-completion-create.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FromSchema, JSONSchema } from 'json-schema-to-ts';
+
+import { chatCompletionSchema } from './chat-completion';
+
+export const chatCompletionCreateBodySchema = {
+  type: 'object',
+  required: ['messages'],
+  properties: {
+    model: { type: 'string' },
+    messages: {
+      type: 'array',
+      items: {
```

Comment:
We can use an enum for the role instead of the `oneOf`.

Reply:
I opted for oneOf because the messages can have different structures based on the role, eventually: https://github.com/openai/openai-openapi/blob/master/openapi.yaml#L11157

Reply:
I see I have to make some further adjustments though 👍